sourcegen-generator-java/src/main/java/io/micronaut/sourcegen/JavaPoetSourceGenerator.java

```diff
@@ -581,6 +578,36 @@ private CodeBlock renderExpression(@Nullable ObjectDef objectDef, MethodDef meth
                 renderExpression(objectDef, methodDef, condition.right())
             );
         }
+        if (expressionDef instanceof ExpressionDef.And andExpressionDef) {
+            boolean isLeftCondition = andExpressionDef.left() instanceof ExpressionDef.Condition ||
+                andExpressionDef.left() instanceof ExpressionDef.And || andExpressionDef.left() instanceof ExpressionDef.Or;
+            boolean isRightCondition = andExpressionDef.right() instanceof ExpressionDef.Condition ||
+                andExpressionDef.right() instanceof ExpressionDef.And || andExpressionDef.right() instanceof ExpressionDef.Or;
+            return CodeBlock.concat(
+                (isLeftCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, andExpressionDef.left()),
+                (isLeftCondition) ? CodeBlock.of(")") : CodeBlock.of(""),
+                CodeBlock.of(" && "),
+                (isRightCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, andExpressionDef.right()),
+                (isRightCondition) ? CodeBlock.of(")") : CodeBlock.of("")
+            );
+        }
+        if (expressionDef instanceof ExpressionDef.Or orExpressionDef) {
+            boolean isLeftCondition = orExpressionDef.left() instanceof ExpressionDef.Condition ||
+                orExpressionDef.left() instanceof ExpressionDef.And || orExpressionDef.left() instanceof ExpressionDef.Or;
+            boolean isRightCondition = orExpressionDef.right() instanceof ExpressionDef.Condition ||
+                orExpressionDef.right() instanceof ExpressionDef.And || orExpressionDef.right() instanceof ExpressionDef.Or;
+            return CodeBlock.concat(
+                (isLeftCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, orExpressionDef.left()),
+                (isLeftCondition) ? CodeBlock.of(")") : CodeBlock.of(""),
+                CodeBlock.of(" || "),
+                (isRightCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, orExpressionDef.right()),
+                (isRightCondition) ? CodeBlock.of(")") : CodeBlock.of("")
```

Comment:
split this into multiple `if/else` to make this easier to read and/or consider rewriting to use switch expressions that assign the code block